nevergrad/functions/images/image_losses.py

```diff
@@ -0,0 +1,46 @@
+import numpy as np
+from koncept.models import Koncept512
+
+
+class _ImageLoss:
+    def __init__(self) -> None:
+        pass
+
+    def compute_loss(self, img: np.ndarray) -> float:
+        assert False
+
+
+class _ImageLossUsingRef(_ImageLoss):
+    def __init__(self, reference: np.ndarray) -> None:
+        super().__init__()
+        self.reference = reference
```

Comment:
If there is no image loss without ref, then no need to define one
More generally, an image loss should just be a function `func(x: np.ndarray) -> float`: right?

In this case, it's more natural to use `__call__` instead of `compute_loss`

I would do something like this:
```python
class ImageLoss
    
    def __init__(self, reference: np.ndarray, method: str) -> None:
         ...

    def __call__(self, x: np.ndarray)-> float:
        ...
```
That will be way shorter and work just fine as well

Reply:
The Koncept512 Loss is an image loss without ref. 
I'll use __call__ instead of compute_loss.

Reply:
All depends in how you want to use this I guess. For benchmarking with nevergrad, having a instance of a custom class as parameter makes things a bit hard because we can't store this object as a descriptor for the benchmark, so if you change it nevergrad won't notice it and you may end up summing xps with different losses without noticing. So for this it's better to avoid such parameters

If you plan on using the ExperimentFunction on your own, out of nevergrad benchmark system, and with different custom losses, then it makes to leave the option open, but we'll need to improve this a bit so that we don't have issues when running benchmarks